nodes/core/io/21-httpin.js

```diff
@@ -73,7 +74,31 @@ module.exports = function(RED) {
                 corsHandler = cors(RED.settings.httpNodeCors);
                 RED.httpNode.options(this.url,corsHandler);
             }
-
+            
+            if(RED.settings.metricsOn && RED.settings.metricsOn === true) {
```

Comment:
Unfortunately I can't get my head around the initialization of the middleware. This currently is an APP level middleware. However what happens is that if the user has more than a single HTTP In node, then the app level middleware keeps getting overwritten for each instance so only the last one to initialize gets kept and used, therefore the metrics wouldn't be reported for every HTTP in node.

I'm thinking about two workarounds:

* One of them would be to use "router" level middleware, I think there I can map the URI to the middleware on a 1 to 1 basis. However I'm unsure if getting a global router singleton is a good idea?

* Second workaround could be an associative array in which I could store the generated middleware functions and I could keep initializing the array of functions to be used. This is a messy workaround as the app.use() would be re-called on each http-in node with more and more function arguments.

For a single HTTP input node, this solution works.